core/templates/dev/head/domain/story/StoryValidationService.js

```diff
@@ -0,0 +1,214 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to validate the consistency of a story. These
+ * checks are performable in the frontend to avoid sending a potentially invalid
+ * story to the backend, which performs similar validation checks to these
+ * in story_domain.Story and subsequent domain objects.
+ */
+
+oppia.factory('StoryValidationService', [
+  'NODE_ID_PREFIX', 'StoryContentsObjectFactory', 'StoryNodeObjectFactory',
+  'StoryObjectFactory',
+  function(
+      NODE_ID_PREFIX, StoryContentsObjectFactory, StoryNodeObjectFactory,
+      StoryObjectFactory) {
+    var _checkValidNodeId = function(nodeId) {
+      if (typeof nodeId !== 'string') {
+        return false;
+      }
+      var nodeIdPattern = new RegExp(NODE_ID_PREFIX + '[0-9]+', 'g');
+      if (!nodeId.match(nodeIdPattern)) {
+        return false;
+      }
+      return true;
+    };
+
+    var _validateNode = function(node) {
+      var issues = [];
+
+      if (!_checkValidNodeId(node.getId())) {
+        throw Error('Invalid node id');
+      }
+      var prerequisiteSkillIds = node.getPrerequisiteSkillIds();
+      var acquiredSkillIds = node.getAcquiredSkillIds();
+      var destinationNodeIds = node.getDestinationNodeIds();
+
+      if ((new Set(prerequisiteSkillIds)).size !==
+          prerequisiteSkillIds.length) {
+        issues.push('All prerequisite skill ids should be distinct');
```

Comment:
Make it easy for the user to fix the issue; specify which skill ID is repeated. Ditto below and elsewhere.

Reply:
Done.